packages/siopv2-oid4vp-op-auth/src/session/functions.ts

```diff
@@ -130,6 +127,79 @@ export async function createOPBuilder({
   return builder
 }
 
+//fixme: this is written based on OID4VCIHolder.signCallback sync the fixes to this with that function
+export function signCallback<T extends JwtIssuer & JwtIssuanceContextBase>(
+  idOpts: ManagedIdentifierOpts,
+  context: IRequiredContext,
+): (jwt: { header: JwtHeader; payload: JwtPayload }, kid?: string) => Promise<string> {
+  return async (jwt: { header: JwtHeader; payload: JwtPayload }, kid?: string) => {
+    let iss = jwt.payload.iss
+    const jwk = jwt.header.jwk
+
```

Comment:
You should not have to do anything of the VCI flow here. The original signer get its arguments pretty much as it should be, so you basically should call the compactJwsSign function. and not the DIDSign. That is the whole point of what you should change

Reply:
refactored the function to use `jwtCreateJwsCompactSignature` function

Reply:
Why is all this code back again?

Reply:
not sure what your question is. code has been refactored and now I am using the `identifierManagedGet` to get a `resolution` for my `idOpts` and at the end calling `jwtCreateJwsCompactSignature` for signing the payload.
```
    const jwk = jwt.header.jwk

    const resolution = await context.agent.identifierManagedGet(idOpts)
    const issuer = jwt.payload.iss || (isManagedIdentifierDidResult(resolution) ? resolution.did : resolution.issuer)
    if (!issuer) {
      return Promise.reject(Error(`No issuer could be determined from the JWT ${JSON.stringify(jwt)}`))
    }
    kid = resolution.kid
    if (kid && isManagedIdentifierDidResult(resolution) && !kid.startsWith(resolution.did)) {
      // Make sure we create a fully qualified kid
      const hash = kid.startsWith('#') ? '' : '#'
      kid = `${resolution.did}${hash}${kid}`
    }
    const header = { ...jwt.header, ...(kid && !jwk && { kid }) } as Partial<JWTHeader>
    const payload = { ...jwt.payload, ...(issuer && { iss: issuer }) }
    if (jwk && header.kid) {
      delete header.kid
    }

    const result: JwsCompactResult = await context.agent.jwtCreateJwsCompactSignature({
      issuer: { identifier: issuer, noIdentifierInHeader: false },
      protectedHeader: header,
      payload,
    })
    return result.jwt
```

Reply:
Yes but I mentioned that you should not copy the kid stuff from elsewhere. That was once removed, now it is back